src/components/legend/draw.js

```diff
@@ -47,365 +47,415 @@ module.exports = function draw(gd) {
     if(typeof gd.firstRender === 'undefined') gd.firstRender = true;
     else if(gd.firstRender) gd.firstRender = false;
 
-    var legend = fullLayout._infolayer.selectAll('g.legend')
-        .data([0]);
+    var legend = createRootNode(),     // root element (sets legend position)
+        clipPath = createClipPath(),   // legend clip path
+        bg = createBackground(),       // legend background and border
+        scrollBox = createScrollBox(), // contains all the legend traces
+        scrollBar = createScrollBar(), // scroll bar (visible only if needed)
+        groups = createGroups(),       // legend groups
+        traces = createLegendTraces(); // legend traces
```

Comment:
: | I already did something similar [here](https://github.com/plotly/plotly.js/commit/c53303b4ac6b3f3bb38482a4e002863c3afe912e). The main purpose is to document the code.

Shall I revert this change and avoid it in the future?